src/ingest_validation_tools/directory_validator.py

```diff
@@ -72,10 +114,24 @@ def _get_required_allowed(dir_schema: List[Dict]) -> Tuple[List[str], List[str]]
     (['this_is_required'], ['this_is_required', 'this_is_optional'])
 
     """
-    allowed = [item["pattern"] for item in dir_schema]
+    allowed = [item["pattern"] for item in dir_schema if "dependency" not in item]
```

Comment:
Are we weeding out sub-dependencies here? If so, are those not a concern?

Reply:
I don't think that we have sub-dependency situations as of yet. If we do run into that, then we will have to reconsider how we approach this.